router/gin/endpoint.go

```diff
@@ -21,10 +21,19 @@ const requestParamsAsterisk string = "*"
 // HandlerFactory creates a handler function that adapts the gin router with the injected proxy
 type HandlerFactory func(*config.EndpointConfig, proxy.Proxy) gin.HandlerFunc
 
-// EndpointHandler implements the HandleFactory interface using the default ToHTTPError function
+// ErrorResponseWriter writes the string representation of an error into the response body
+// and sets a Content-Type header for errors that implement the encodedResponseError interface.
+var ErrorResponseWriter = func(c *gin.Context, statusCode int, err error) {
+	if te, ok := err.(encodedResponseError); ok && te.Encoding() != "" {
+		c.Header("Content-Type", te.Encoding())
+	}
+	c.String(statusCode, err.Error())
```

Comment:
This should be `c.Writer.WriteString(err.Error())` so no header is added. With this change, the tests will pass

Reply:
thanks! Fixed it now and the tests passed locally